pkg/pillar/base/logobjecttypes.go

```diff
@@ -111,9 +111,15 @@ func NewLogObject(objType LogObjectType, objName string, objUUID uuid.UUID, key
 		log.Fatal("NewLogObject: objType and key parameters mandatory")
 	}
 	// Check if we already have an object with the given key
-	object, ok := logObjectMap[key]
+	var object *LogObject
+	value, ok := logObjectMap.Load(key)
 	if ok {
-		return object
+		object, ok = value.(*LogObject)
+		if ok {
+			return object
+		}
+		// Delete whatever we have right now that is not LogObject and allocate new.
+		DeleteLogObject(key)
```

Comment:
Why would the type assertion ever fail? Shouldn't this be a fatal since it would be an utter bug?

Note that subtle bugs can exist, such as the difference between a pointer and non-pointer type, when the type assertion will fail. We need to find and fix such bugs.
Thus either do object = value.(*LogObject), or have the !ok path do a log Fatal which prints the type using %T.

Reply:
will do